tests/unit/test_board.py

```diff
@@ -80,3 +80,11 @@ def test_board_repr():
     piece = Pawn(direction=-1, position="a2", representation="o1")
     board = Board(pieces=[piece], size=8)
     board.__repr__()
+
+def test_board_fifty_move_draw():
+    bishop = Bishop(direction=-1, position="a1", representation="B1")
+    board = Board(pieces=[bishop], size=8)
+    for i in range(50):
+        board.move_piece(bishop, (1,1))
+        board.move_piece(bishop, (0,0))
+    assert board.is_draw_by_fifty_moves()
```

Comment:
- Final newline missing.
- Variable i not required
- Test interaction with `ReversibleMove` context manager is required to avoid bugged minimax calculations, e.g.:
```python
from chess_ng.algorithm import ReversibleMove

# ...

def test_board_fifty_move_draw():
    bishop = Bishop(direction=-1, position="a1", representation="B1")
    board = Board(pieces=[bishop], size=8)
    for _ in range(49):
        board.move_piece(bishop, (1,1))
        board.move_piece(bishop, (0,0))
    board.move_piece(bishop, (1,1))
    with ReversibleMove(board, bishop, position=(0,0), enemy_pieces=[]):
        assert board.is_draw_by_fifty_moves()
    assert not board.is_draw_by_fifty_moves()
```